lib/rubocop/cop/rspec/capybara/negation_matcher.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      module Capybara
+        # Enforces use of `have_no_*` or `not_to` for negated expectations.
+        #
+        # @example EnforcedStyle: not_to (default)
+        #   # bad
+        #   expect(page).to have_no_selector
+        #   expect(page).to have_no_css('a')
+        #
+        #   # good
+        #   expect(page).not_to have_selector
+        #   expect(page).not_to have_css('a')
+        #
+        # @example EnforcedStyle: have_no
+        #   # bad
+        #   expect(page).not_to have_selector
+        #   expect(page).not_to have_css('a')
+        #
+        #   # good
+        #   expect(page).to have_no_selector
+        #   expect(page).to have_no_css('a')
+        #
+        class NegationMatcher < Base
+          extend AutoCorrector
+          include ConfigurableEnforcedStyle
+
+          MSG = 'Use `%<good>s` instead of `%<bad>s`.'
+          CAPYBARA_MATCHERS = %w[
+            selector css xpath text title current_path link button
+            field checked_field unchecked_field select table
+            sibling ancestor
+          ].freeze
+          POSITIVE_MATCHERS = Set.new(CAPYBARA_MATCHERS.map do |element|
+            "have_#{element}".to_sym
+          end).freeze
+          NEGATIVE_MATCHERS = Set.new(CAPYBARA_MATCHERS.map do |element|
+            "have_no_#{element}".to_sym
+          end).freeze
+          RESTRICT_ON_SEND = (POSITIVE_MATCHERS + NEGATIVE_MATCHERS).freeze
+
+          # @!method not_to?(node)
+          def_node_matcher :not_to?, <<~PATTERN
+            (send ... :not_to
+              (send nil? %POSITIVE_MATCHERS ...))
+          PATTERN
+
+          # @!method have_no?(node)
+          def_node_matcher :have_no?, <<~PATTERN
+            (send ... :to
+              (send nil? %NEGATIVE_MATCHERS ...))
+          PATTERN
+
+          def on_send(node)
+            return unless offense?(node.parent)
+
+            add_offense(offense_range(node),
+                        message: message(node)) do |corrector|
+              corrector.replace(node.parent.loc.selector, replaced_target)
+              corrector.replace(node.loc.selector, replaced_matcher(node))
+            end
+          end
+
+          private
+
+          def offense?(node)
+            (style == :have_no && not_to?(node)) ||
+              (style == :not_to && have_no?(node))
```

Comment:
It seems like the `style` check here, inside `#replaced_target` and inside `#replaced_matcher` is implemented in three different ways – the two latter use “else” instead of an explicit `style == :not_to`.

How about using e.g.
```rb
if style == :have_no
  ...
else
  ...
end
```
or
```rb
case style
when :have_no
  ...
when :not_to
  ...
end
```
in all three places?

Reply:
I unified with the style expressed in case.